pkg/rke2/psp/psp_templates.go

```diff
@@ -0,0 +1,199 @@
+package psp
+
+const (
+	globalUnrestrictedPSPName         = "global-unrestricted-psp"
+	globalUnrestrictedRoleName        = "global-unrestricted-psp-clusterrole"
+	globalUnrestrictedRoleBindingName = "global-unrestricted-psp-rolebinding"
+
+	globalRestrictedPSPName         = "global-restricted-psp"
+	globalRestrictedRoleName        = "global-restricted-psp-clusterrole"
+	globalRestrictedRoleBindingName = "global-restricted-psp-rolebinding"
+
+	systemUnrestrictedPSPName                = "system-unrestricted-psp"
+	systemUnrestrictedRoleName               = "system-unrestricted-psp-role"
+	systemUnrestrictedRoleBindingName        = "system-unrestricted-node-psp-rolebinding"
+	systemUnrestrictedSvcAcctRoleBindingName = "system-unrestricted-svc-acct-psp-rolebinding"
+)
+
+const roleTemplate = `kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: %s
+rules:
+- apiGroups: ['policy']
+  resources: ['podsecuritypolicies']
+  verbs:     ['use']
+  resourceNames:
+  - %s
+`
+
+const globalRestrictedPSPTemplate = `apiVersion: policy/v1beta1
```

Comment:
After seeing how this looks, I think we could consider switching from string constants to just declaring the actual instances structs. It would save on the decoding and provide a bit of type safety.

In my mind, the main justification for string constants would be readability, but I think the structs do not lose enough in terms of readability. Is there another good reason for string constants that would tip the scales?

Sample struct:
```
var falseVal = false
var psp1 = v1beta1.PodSecurityPolicy{
	Spec: v1beta1.PodSecurityPolicySpec{
		Privileged: false,
		AllowPrivilegeEscalation: &falseVal,
		RunAsUser:                       v1beta1.RunAsUserStrategyOptions{
			Rule: v1beta1.RunAsUserStrategyMustRunAsNonRoot,
		},
           ...
```
feels relatively readable (assuming you alrady no go)

Reply:
I don't know. I'm wavering on this. the string constant is probably fine

Reply:
I'm happy to change it. I was just mirroring RKE.

Reply:
nice-to-have. in fact, I think a case could be made either way, so you can leave as-is

Reply:
Issue created.